### DISCUSSION TITLE
Vacuum into the file: `DatabaseException(near "INTO": syntax error (code 1 SQLITE_ERROR): , while compiling: VACUUM INTO ?)`

### DISCUSSION THREAD

**Discussion by madsane29** [simolus3/drift#2068] [Q&A] [answered]
So I am trying to VACUUM the database into a file by using the example at https://drift.simonbinder.eu/docs/examples/existing_databases/#exporting-a-databasee :

```
main() {
  Directory directory = await getApplicationDocumentsDirectory();
  String fileName = "my-file-name";
  File dbFile = await _getDbFile(directory, fileName);
}

_getDbFile(Directory directory, String fileName) async {
    File dbFile = File("${directory.path}/${fileName}.db");
    await _exportInto(dbFile);
    return dbFile;
}

_exportInto(File file) async {
    await file.parent.create(recursive: true);

    if (file.existsSync()) {
      file.deleteSync();
    }

    await appDatabase.customStatement('VACUUM INTO ?', [file.path]);
}
```

But I get the following exception:

```
I/flutter (25951): Drift: Sent VACUUM INTO ? with args [/data/user/0/my-app/app_flutter/my-file-name.db]
E/SQLiteLog(25951): (1) near "INTO": syntax error
E/flutter (25951): [ERROR:flutter/lib/ui/ui_dart_state.cc(198)] Unhandled Exception: DatabaseException(near "INTO": syntax error (code 1 SQLITE_ERROR): , while compiling: VACUUM INTO ?) sql 'VACUUM INTO ?' args [/data/user/0/my-app/app_flutter/my-fi...]
E/flutter (25951): #0      wrapDatabaseException (package:sqflite/src/exception_impl.dart:11:7)
E/flutter (25951): <asynchronous suspension>
E/flutter (25951): #1      BasicLock.synchronized (package:synchronized/src/basic_lock.dart:33:16)
E/flutter (25951): <asynchronous suspension>
E/flutter (25951): #2      SqfliteDatabaseMixin.txnSynchronized (package:sqflite_common/src/database_mixin.dart:344:14)
```

This is my pubspec.yaml:

![image](https://user-images.githubusercontent.com/40585482/191446253-8997d29e-b0b0-400a-9eff-f73357d8dad1.png)

I wonder if anyone knows what is the problem?

Thanks in advance.

**Comment by simolus3** [comment, marked as answer]
Support for `VACUUM INTO` has been added in sqlite3 version 3.37.0.

Since you're using `moor_flutter`, which in turn uses `sqflite` which uses the sqlite3 library that comes with the operating system, this may not work on some devices coming with an older sqlite3 version.
If you really need this feature, I recommend using a `NativeDatabase` from `package:moor/ffi.dart` instead of a `FlutterQueryExecutor` as described [here](https://drift.simonbinder.eu/docs/other-engines/vm/#migrating-from-moor_flutter-to-moor-ffi). If you want to keep using a `FlutterQueryExecutor`, you'll have to accept that you get different sqlite3 versions on different devices.

**Reply by madsane29** [reply]
Alright, I migrated to drift from moor and used the link you sent me and it is working great. Now the project is up to date in this package's perspective. Thanks!
